### DISCUSSION TITLE
React compatibility

### DISCUSSION THREAD

**Discussion by ghost** [asmcss/assembler#26] [Q&A]
How Assembler CSS will compatible with React? As I know React uses JS objects in style property.

**Comment by msarca** [comment]
The way React, Vue, and similar frameworks handle the `style` and `class` attributes is very problematic. Historically, these attributes belong to everyone, and no framework should have exclusive rights over them. One could use a [custom selector attribute](https://asmcss.com/docs/1.x/configuration.html#data-selector-attribute), but this doesn't solve the problem with `style`, because `style` is a special HTML attribute an it can not be replaced. So using a `style` and an `x-style` for the same element is not recommended when using React because of the way it handle the diff. This is not something specific to Assembler CSS, using a custom element with React can create the same conflict.
